### PR TITLE
 JS: introduce 'Util::describeExpression'

### PR DESCRIPTION
**Pull request by esben-semmle**
As discussed recently, it may be clearer to mention the kind of an expression in an alert message, instead of just using the generic "this expression is bad".

This PR introduce utility for doing just that: `Util::describeExpression`. I have put it to use in `js/trivial-conditional`, we can use it in other queries later.

The [evalution](https://git.semmle.com/esben/dist-compare-reports/tree/js/improve-useless-conditional-message_1540982810831) shows that performance does not degrade.

Ping @max as he may have an opinion on this.

### CODE REVIEW

**Comment by esben-semmle** [comment]
Not that @max. Sorry about that!
Ping @xiemaisi

File: javascript/ql/src/Statements/UselessConditional.ql [hunk 1]
```diff
@@ -134,10 +134,13 @@ where guard.getTest() = op.asExpr() and
...
-        if sel instanceof VarAccess then
-          msg = "Variable '" + sel.(VarAccess).getVariable().getName() + "' always evaluates to " + cv + " here."
-        else
-          msg = "This expression always evaluates to " + cv + "."
+        exists (string description |
+          description = describeExpression(sel) |
+          if sel instanceof VarAccess then
+            msg = capitalize(description) + " always evaluates to " + cv + " here."
+          else
+            msg = "This " + description + " always evaluates to " + cv + "."
```
Comment (asger-semmle) [inline javascript/ql/src/Statements/UselessConditional.ql]: In general we could get messages like:
```
This property 'f' always evaluates to true.
```
In general I think "named" expressions should belong in the former category.
```
Property 'f' always evaluates to true here.
```
We could add a boolean flag to `describeExpression` indicating whether it is named.

Comment (esben-semmle) [reply javascript/ql/src/Statements/UselessConditional.ql]: The problem is the two kinds of messages. How about we conflate them to the latter, with variables being described as "read of variable 'v'":

Examples:
```
This comparison always evaluate to true.
This expression always evaluate to true.
This read of variable 'v' always evaluate to true.
This read of property 'p' always evaluate to true.
This call to 'f' always evaluate to true.
This property read always evaluate to true.
This call always evaluate to true.
```

Comment (asger-semmle) [reply javascript/ql/src/Statements/UselessConditional.ql]: I would prefer `This use of` instead of `This read of`.

Comment (esben-semmle) [reply javascript/ql/src/Statements/UselessConditional.ql]: Done + some additional features. I am not sure I like messages with the callee name though: they are long, and the callee name is not a link.

**Comment by esben-semmle** [comment]
Rebased.

**Pull request merged**